docs architecture: dedicated SnarkyJS section (#384)
o1-labs/docs2: docs/zkapps/how-to-write-a-zkapp.mdx

If you haven’t yet read the [how zkApps work](how-zkapps-work) pages, please read it now so that this section makes sense.

> ##### SnarkyJS

Author: this content now lives snarkyjs-overview.mdx

### Writing your smart contract

zkApps are written in TypeScript using SnarkyJS. SnarkyJS is a TypeScript library for writing smart contracts based on zero-knowledge proofs for the Mina Protocol. It is included automatically when creating a new project using the Mina zkApp CLI.

See [the SnarkyJS docs](snarkyjs) to learn more. The first 3 subsections are enough to get you started writing zkApps:

- [Basic concepts](./snarkyjs/basic-concepts)
- [Smart contracts](./snarkyjs/smart-contracts)
- [Interacting with Mina](./snarkyjs/interact-with-mina)

> Alternatively, we recommend [Tutorial 1: Hello World](./tutorials/hello-world) as a hands-on walkthrough that teaches you to create your first zkApp.

Author: ```suggestion
Start with [Tutorial 1: Hello World](./tutorials/hello-world), a hands-on walkthrough that teaches you to create your first zkApp.
```
